### ISSUE TITLE
`seq` data training never occurs in the code

### ISSUE THREAD

**Issue by hiteshvaidya** [Orienfish/SCALE#3]
Hello,

I read the paper and went through the code and realized that the sequential data training never occurs in 
 [main_supcon.py](https://github.com/Orienfish/SCALE/blob/main/OnlineContrast/main_supcon.py). `run_scale.sh` sets `--training_data_type` to `class_iid` when we set data stream as 'seq'. However, this `training_data_type` is never used in the code. 

As per the paper, 'seq' setting passes the samples in a class incremental when we set `seq` data stream. However, the dataloader loads data in iid form and does not train the network on a single sequential class number (eg. 0,1,2,...9). Therefore, even the validation function evaluates the model on iid data samples and does not generate any task matrix like reported in many continual/lifelong learning papers [GEM](https://proceedings.neurips.cc/paper_files/paper/2017/file/f87522788a2be2d171666752f97ddebb-Paper.pdf).

**Comment by Orienfish** [comment]
Thanks for taking a look at our work!

The `opt.training_data_type` is used in `data_utils.py` in L312 [[here](https://github.com/Orienfish/SCALE/blob/main/data_utils.py#L312)]. The `data_utils.py` is responsible for configuring the training and validation data. If you specify `opt.training_data_type` to be `seq`, it will lead to different configuration for the data loader in this file.

For your question re. the validation function, our validation function evaluates on the iid data samples, and is different from the task-based continual learning papers. The Section 3 "training and evaluation protocol" in the SCALE paper provides a detailed explanation. 

I'll leave this issue open for a week. Please let me know if you have any follow-up questions.

**Comment by hiteshvaidya** [comment]
I want to obtain a task matrix of evaluation accuracies of SCALE and further check other metrics like Backward Transfer and Learning Accuracy. In order to obtain that, I guess I will have to change the [val_loader](https://github.com/Orienfish/SCALE/blob/dc095da25db5e44d702062f77ec35e5578038c2c/data_utils.py#L333) and [knn_train_loader](https://github.com/Orienfish/SCALE/blob/dc095da25db5e44d702062f77ec35e5578038c2c/data_utils.py#L340) to return task/class wise samples?

If I am not mistaken, the [knn_task_eval()](https://github.com/Orienfish/SCALE/blob/dc095da25db5e44d702062f77ec35e5578038c2c/eval_utils.py#L259) does perform task wise evaluation but it calculates an average accuracy over samples from all tasks and not give a task-wise accuracy vector? The [eval_forget()](https://github.com/Orienfish/SCALE/blob/dc095da25db5e44d702062f77ec35e5578038c2c/eval_utils.py#L290) function does calculate forgetting measure but is never referenced in the code.

**Comment by Orienfish** [comment]
Yes, your understands are correct.
